pdf/model/textencoding/encoder.go

```diff
@@ -50,20 +53,45 @@ type TextEncoder interface {
 
 // Convenience functions
 
-// doEncode converts a Go unicode string `raw` to a PDF encoded string using the encoder `enc`.
-func doEncode(enc TextEncoder, raw string) []byte {
-	encoded := []byte{}
+// doEncode1 converts a Go unicode string `raw` to a PDF encoded string using the encoder `enc`.
+// It expects that character codes will fit into a single byte.
+func doEncode1(enc TextEncoder, raw string) []byte {
+	encoded := make([]byte, 0, len(raw))
 	for _, r := range raw {
 		code, found := enc.RuneToCharcode(r)
 		if !found {
 			common.Log.Debug("Failed to map rune to charcode for rune 0x%04x", r)
 			continue
 		}
+		if code > 0xff {
+			panic(fmt.Errorf("unexpected character code: %x", code))
```

Comment:
Please don't panic :).  PDF is full of surprises so don't want the program to crash as a result of bad input.
Would be better if we can pass the error up (return err).  Otherwise just log the error (common.Log.Debug) and break out.

Reply:
Added a note in the style guide on this
https://github.com/unidoc/unidoc/wiki/UniDoc-Developer-Guide#16-error-handling---dont-panic